app/src/main/java/com/google/android/gms/samples/wallet/PaymentsUtil.kt

```diff
@@ -0,0 +1,272 @@
+     */
+    private fun cardPaymentMethod(): JSONObject {
+        val cardPaymentMethod = baseCardPaymentMethod()
+        cardPaymentMethod.put("tokenizationSpecification", gatewayTokenizationSpecification())
+
+        return cardPaymentMethod
+    }
+
+    /**
+     * An object describing accepted forms of payment by your app, used to determine a viewer's
+     * readiness to pay.
+     *
+     * @return API version and payment methods supported by the app.
+     * @see [IsReadyToPayRequest](https://developers.google.com/pay/api/android/reference/object.IsReadyToPayRequest)
+     */
+    fun isReadyToPayRequest(): Optional<JSONObject> {
+        try {
+            val isReadyToPayRequest = baseRequest
+            isReadyToPayRequest.put(
+                    "allowedPaymentMethods", JSONArray().put(baseCardPaymentMethod()))
+
+            return Optional.of(isReadyToPayRequest)
+        } catch (e: JSONException) {
+            return Optional.empty()
+        }
+    }
+
+    /**
+     * Information about the merchant requesting payment information
+     *
+     * @return Information about the merchant.
+     * @throws JSONException
+     * @see [MerchantInfo](https://developers.google.com/pay/api/android/reference/object.MerchantInfo)
+     */
+    private val merchantInfo: JSONObject
+        @Throws(JSONException::class)
+        get() = JSONObject().put("merchantName", "Example Merchant")
+
+    /**
+     * Creates an instance of [PaymentsClient] for use in an [Activity] using the
+     * environment and theme set in [Constants].
+     *
+     * @param activity is the caller's activity.
+     */
+    fun createPaymentsClient(activity: Activity): PaymentsClient {
+        val walletOptions = Wallet.WalletOptions.Builder().setEnvironment(Constants.PAYMENTS_ENVIRONMENT).build()
+        return Wallet.getPaymentsClient(activity, walletOptions)
+    }
+
+    /**
+     * Provide Google Pay API with a payment amount, currency, and amount status.
+     *
+     * @return information about the requested payment.
+     * @throws JSONException
+     * @see [TransactionInfo](https://developers.google.com/pay/api/android/reference/object.TransactionInfo)
+     */
+    @Throws(JSONException::class)
+    private fun getTransactionInfo(price: String): JSONObject {
+        val transactionInfo = JSONObject()
+        transactionInfo.put("totalPrice", price)
+        transactionInfo.put("totalPriceStatus", "FINAL")
+        transactionInfo.put("currencyCode", Constants.CURRENCY_CODE)
+
+        return transactionInfo
+    }
+
+    /**
+     * An object describing information requested in a Google Pay payment sheet
+     *
+     * @return Payment data expected by your app.
+     * @see [PaymentDataRequest](https://developers.google.com/pay/api/android/reference/object.PaymentDataRequest)
+     */
+    fun getPaymentDataRequest(price: String): Optional<JSONObject> {
```

Comment:
I know we now have Optional as part of Java 8 but I feel it would feel more of a kotlin approach to just use nullable return types. I might be missing something about keeping the use of Optional here so let me know otherwise!

Reply:
Removing `optional` would also allow us to reduce the minsdk version to 19, which is what the GPay SDK supports down to

Reply:
💯

Reply:
If you remove org.json then no need to handle this case

Reply:
Replaced Optionals with nullables and increased minSdk to 19. Replacing JSON parsing is out of scope of this PR.